### PR TITLE
More gc enforcement policies

### PR DESCRIPTION
**Pull request by spring2maz**

### CODE REVIEW

**Comment by spring2maz** [comment]
rebase

File: priv/emqx.schema [hunk 1]
```diff
@@ -1750,3 +1750,25 @@ end}.
...
+%% @doc Force connection/session process GC after this number of
+%% bytes passed through
+{mapping, "mqtt.conn.force_gc_bytes", "emqx.conn_force_gc_bytes", [
+  {datatype, integer}
+]}.
+
+%% @doc Minimal interval in seconds to force connection/session process GC
+{mapping, "mqtt.conn.force_gc_interval", "emqx.conn_force_gc_interval", [
+  {datatype, integer}
+]}.
```
Comment (emqplus) [inline priv/emqx.schema]: Combine the three options to one 'zone.<name>.force_gc_policy':
```
{mapping, "zone.$name.force_gc_policy", "emqx.conn_force_gc_count", [
   {default, "Interval|Count|Bytes"},
   {datatype, string}
 ]}.

Comment (emqplus) [reply priv/emqx.schema]: Translate the 'force_gc_policy' to a map:

```
lists:to_map(lists:zip([interval, count, bytes],
                                    [list_to_integer(Token) || Token <- string:tokens("Interval|Count|Bytes", "|")]))
```

**Comment by spring2maz** [comment]
fixed review comments.

**Comment by emqplus** [comment]
Merge the PR first. The `emqx_gc` should be a parameterized module.

**Comment by spring2maz** [comment]
parameterized module is not supported (not officially) since otp 21
do we really want to add it with an ugly hack ?

**Pull request merged**